File: haskell/problem03.hs
import Data.List

main = print $ largestPrimeFactor 600851475143

largestPrimeFactor x =
    let largest = floor . sqrt $ fromIntegral x
        smallest = find ((==0) . mod x) [2..largest]
    in  case smallest of
            Nothing -> x
            Just y -> largestPrimeFactor (x `div` y)

File: javascript/problem03.js
function largestPrimeFactor(num) {
  var maxVal = Math.floor( Math.sqrt(num) );
  for (i = 2; i < maxVal ; i++) {
    if (num % i === 0) {
      return largestPrimeFactor(num / i);
    }
  }

  return num;
}

console.log( largestPrimeFactor(6008514751) );
